cmd/maya-apiserver/cstor-operator/cspc/operations.go

```diff
@@ -74,28 +75,40 @@ func (pc *PoolConfig) expandPool() error {
 }
 
 // addGroupToPool adds a raid group to the csp
-func addGroupToPool(cspcPoolSpec *apis.PoolSpec, csp *apis.NewTestCStorPool) *apis.NewTestCStorPool {
+func (pc *PoolConfig) addGroupToPool(cspcPoolSpec *apis.PoolSpec, csp *apis.NewTestCStorPool) *apis.NewTestCStorPool {
 	for _, cspcRaidGroup := range cspcPoolSpec.RaidGroups {
+		validGroup := true
 		cspcRaidGroup := cspcRaidGroup
 		if !isRaidGroupPresentOnCSP(&cspcRaidGroup, csp) {
 			if cspcRaidGroup.Type == "" {
 				cspcRaidGroup.Type = cspcPoolSpec.PoolConfig.DefaultRaidGroupType
 			}
-			csp.Spec.RaidGroups = append(csp.Spec.RaidGroups, cspcRaidGroup)
+			for _, bd := range cspcRaidGroup.BlockDevices {
```

Comment:
how about calling `ClaimBDsForNode` by passing array of BD names.. `ClaimBDsForNode` can be renamed to `ClaimBDs`

Reply:
or rename it to `AreBDsClaimed`